packages/cozy-harvest-lib/src/components/OAuthForm.jsx

```diff
@@ -16,142 +16,127 @@ import { KonnectorJobError } from '../helpers/konnectors'
-        : 'oauth.reconnect.label'
-      : isBankingKonnector
-      ? 'oauth.banking.connect.label'
-      : 'oauth.connect.label'
-
-    return (
-      <>
-        {error && (
-          <TriggerErrorInfo
-            className="u-mb-1"
-            error={error}
-            konnector={konnector}
-          />
-        )}
-        {!reconnect && (
-          <Button
-            className="u-mt-1"
-            busy={isBusy}
-            disabled={isBusy}
-            extension="full"
-            label={t(buttonLabel)}
-            onClick={this.handleConnect}
-          />
-        )}
-        {showOAuthWindow && (!needExtraParams || extraParams) && (
-          <OAuthWindow
-            extraParams={extraParams}
-            konnector={konnector}
-            reconnect={reconnect}
-            onSuccess={this.handleAccountId}
-            onCancel={this.handleOAuthCancel}
-            account={account}
-            intentsApi={intentsApi}
-          />
-        )}
-      </>
-    )
+  const handleLoginSuccess = () => {
+    setShowOAuthWindow(false)
   }
+
+  useEffect(() => {
+    const konnectorPolicy = findKonnectorPolicy(konnector)
+
+    if (konnectorPolicy.fetchExtraOAuthUrlParams) {
+      setNeedsExtraParams(true)
+      if (reconnect) {
+        setShowOAuthWindow(true)
+      }
+      // eslint-disable-next-line promise/catch-or-return
+      konnectorPolicy
+        .fetchExtraOAuthUrlParams({
+          account,
+          konnector,
+          client,
+          reconnect
+        })
+        .then(setExtraParams)
+    }
+    flow.on(LOGIN_SUCCESS_EVENT, handleLoginSuccess)
+
+    return () => {
+      flow.removeListener(LOGIN_SUCCESS_EVENT, handleLoginSuccess)
+    }
+  }, [])
+
+  const { error } = flowState
+  const isBusy =
+    showOAuthWindow || flowState.running || (needsExtraParams && !extraParams)
+  const isBankingKonnector = konnector.categories?.includes('banking')
+  const buttonLabel = reconnect
+    ? isBankingKonnector
+      ? 'oauth.banking.reconnect.label'
+      : 'oauth.reconnect.label'
+    : isBankingKonnector
+    ? 'oauth.banking.connect.label'
+    : 'oauth.connect.label'
+
+  return (
+    <>
+      {error && (
+        <TriggerErrorInfo
+          className="u-mb-1"
+          error={error}
+          konnector={konnector}
+        />
+      )}
+      {!reconnect && (
+        <Button
+          className="u-mt-1"
+          busy={isBusy}
+          disabled={isBusy}
+          extension="full"
+          label={t(buttonLabel)}
+          onClick={handleConnect}
+        />
+      )}
+      {showOAuthWindow && (!needsExtraParams || extraParams) && (
+        <OAuthWindow
+          extraParams={extraParams}
+          konnector={konnector}
+          reconnect={reconnect}
+          onSuccess={handleAccountId}
+          onCancel={handleOAuthCancel}
+          account={account}
+          intentsApi={intentsApi}
+        />
+      )}
+    </>
+  )
 }
 
 OAuthForm.propTypes = {
```

Comment:
@taratatach nit FYI this should not be tagged as `chore`. `Chore` is only about stuff that are not deployed like CI stuff or things like that. Here you are editing what we consume so it's more a `refactor` commit https://github.com/cozy/cozy-guidelines/#type